c/cert/test/rules/SIG30-C/test.c

```diff
@@ -0,0 +1,122 @@
+#include <signal.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+char *info = NULL;
+
+void log_local_unsafe(void) { fputs(info, stderr); }
+
+void handler1(int signum) {
```

Comment:
I'm a little confused by the location of the COMPLIANT and NON_COMPLIANT markers. For example, I'm not sure why the call to signal in f1 is compliant. It looks like the OTHER instances of the markers are on the unsafe thing. In this example, signal may eventually invoke handler1 which may call log_local_unsafe which in turn calls unsafe fputs. Can you explain the logic here?